pkg/controller/elasticsearch/client/base.go

```diff
@@ -129,7 +131,18 @@ func (c *baseClient) request(
 		body = bytes.NewBuffer(outData)
 	}
 
-	request, err := http.NewRequest(method, stringsutil.Concat(c.Endpoint, pathWithQuery), body) //nolint:noctx
+	rawURL := stringsutil.Concat(c.Endpoint, pathWithQuery)
+	if c.debug {
+		url, err := url.Parse(rawURL)
+		if err != nil {
+			return err
+		}
+		q := url.Query()
+		q.Set("error_trace", "true")
+		url.RawQuery = q.Encode()
+		rawURL = url.String()
+	}
+	request, err := http.NewRequest(method, rawURL, body) //nolint:noctx
 	if err != nil {
```

Comment:
Why don't you wait to check to see if debug is enabled until after you generate the request? Then you won't have to parse the URL and have that additional error logic.  Something like

```
q := request.URL.Query()
q.Add("error_trace", "true")
request.URL.RawQuery = q.Encode()
```